peer/peer.go

```diff
@@ -80,8 +82,16 @@ func (p *Peer) Gossip(ctx context.Context, contentID []byte, subnet *id.Hash) {
 	p.gossiper.Gossip(ctx, contentID, subnet)
 }
 
+func (p *Peer) PeerDiscovery(ctx context.Context) {
+	p.discoveryClient.DiscoverPeers(ctx)
+}
```

Comment:
This method should not exist. Instead, we should run `DiscoverPeers` during the `Peer.Run` method, because `DiscoverPeers` is a method that runs constantly in the background. Then, the `Peer.Run` function would look like something like this:

```
	// ... other stuff

	var wg sync.WaitGroup
	wg.Add(2)
	go func() {
		defer wg.Done()
		p.discoveryClient.DiscoverPeers(ctx)
	}()
	go func() {
		defer wg.Done()
		p.transport.Run(ctx)
	}()
	wg.Wait()
```

Reply:
This will actually cause problems. Peer discovery can only be run if `p.transport.Run` has started accepting messages. Starting them at the same time will cause peer discovery to throw errors. Secondly, a user might be bootstrapping and might not want to run peer discovery anyway. I think we should leave the responsibility on the user